modules/storage/glance.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2022 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// GlanceExtraVolTypeUndefined can be used to label an extraMount which
+	// is not associated with a specific backend
+	GlanceExtraVolTypeUndefined ExtraVolType = "Undefined"
+	// GlanceExtraVolTypeCeph can be used to label an extraMount which
+	// is associated to a Ceph backend
+	GlanceExtraVolTypeCeph      ExtraVolType = "Ceph"
+
+	// GlanceAPI defines the glance-api group
+	GlanceAPI ServiceType = "GlanceAPI"
+
+	// Glance is the global ServiceType that refers to all the components deployed
+	// by the glance operator
+	Glance ServiceType = "Glance"
+)
+
+// GlanceExtraVolMounts exposes additional parameters processed by the glance-operator
+// and defines the common VolMounts structure provided by the main storage module
+type GlanceExtraVolMounts struct {
+	// +kubebuilder:validation:Optional
+	Name string `json:"name,omitempty"`
+	// +kubebuilder:validation:Optional
+	Region string `json:"region,omitempty"`
+	// +kubebuilder:validation:Required
+	VolMounts []VolMounts `json:"extraVol"`
```

Comment:
So just to see if I understood the intention properly. A single VolMounts is a list of Volumes&Mounts with a specific propagation rule set. So with two VolMounts here we can have two sets of Volumes&Mounts that are propagated by different rules. Is it so?

Reply:
Exactly, your understanding is correct:
```
...
...
  extraMounts:
      extraVol:
        - propagation:
          - Cinder
          - Compute
          volumes:
          - name: ceph
            projected:
              sources:
              - secret:
                  name: ceph-client-conf
              - secret:
                  name: ceph-client-conf2
          mounts:
          - name: ceph
            mountPath: "/etc/ceph"
            readOnly: true
        - propagation:
          - Glance
          - Compute
          extraVolType: Ceph
          volumes:
          - name: ceph2
            projected:
              sources:
              - secret:
                  name: ceph-client-conf
          mounts:
          - name: ceph2
            mountPath: "/etc/ceph2"
            readOnly: true
```

Two set of volumes with two different propagation rules.